lib/storage_tables/attachable/one.rb

```diff
@@ -14,8 +14,12 @@ class One < ActiveStorage::Attached::One
       #   person.avatar.attach(params[:signed_blob_id]) # Signed reference to blob from direct upload
       #   person.avatar.attach(io: File.open("/path/to/face.jpg"), filename: "face.jpg", content_type: "image/jpeg")
       #   person.avatar.attach(avatar_blob) # ActiveStorage::Blob object
-      def attach(attachable, filename:)
-        record.public_send("#{name}=", attachable, filename)
+      def attach(attachable, filename: nil)
+        filename ||= extract_filename(attachable)
+
+        raise ArgumentError, "Could not determine filename from #{attachable.inspect}" unless filename
+
+        record.public_send(:"#{name}=", attachable, filename)
         blob.save! && upload(attachable)
         # :nocov:
```

Comment:
Dit :nocov: doet helemaal niks in SonarCloud, dit is van de originele implementatie?

Reply:
Dit doet sowieso niks met SonarCloud, ik heb dit toegevoegd. Omdat dit scenario eigenlijk niet te testen is. De check is overgenomen vanuit ActiveStorage. Maar eigenlijk zijn er al meerdere checks voor die zorgen dat dit scenario niet voorkomt. Ik heb het wel erin gelaten, maar was in orgineel ook niet gecovered door een test.

Reply:
Check